### ISSUE TITLE
Lint on the fly doesn't work with elm-test

### ISSUE THREAD

**Issue by opsb** [mybuddymichael/linter-elm-make#115]
By default the `tests/elm-package.json` points to the src directory in the parent project. This causes the warning

![image](https://cloud.githubusercontent.com/assets/46232/22182683/c33aeafc-e0ab-11e6-87ea-db9e81b43b98.png)

Is it possible to loosen the constraint to allow for this scenario or is there a fundamental issue?

**Comment by halohalospecial** [comment]
Hi!  Currently, `Lint On The Fly` works by copying source .elm files to the work directory (which can be a temporary directory if not set).  If a source directory is outside the project directory, scary things might happen like unintentionally modifying or deleting your other files.  Suggestions on how to solve this are most welcome :)

@splodingsocks, can you still remember the specific issue that you encountered regarding this?

**Comment by mrmurphy** [comment]
Yeah, I believe the issue here is that `elm-test`'s default configuration makes your test files act like a totally different Elm project (since it's got its own `elm-package.json`. So this isn't so much of an issue with `elm-test` alone, but an issue whenever you're trying to use one separate Elm project from within another.

I haven't used lint-on-the-fly for a long time for this reason. It'd be nice to have working, though!

**Comment by nickdima** [comment]
Someone has a solution for this yet?

**Comment by samgqroberts** [comment]
I've been able to work around this issue by collocating my tests with my main modules. (Obviously this is only if you are okay with doing that, I personally prefer it). I followed the conversation here to get my tests under `src/` (and remove the `tests/` dir altogether) https://github.com/rtfeldman/node-test-runner/issues/241
